app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -57,22 +59,24 @@ import com.infomaniak.mail.utils.WebViewUtils.Companion.toggleWebViewTheme
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import java.util.*
+import java.util.Date
 import com.google.android.material.R as RMaterial
 
 class ThreadAdapter(
     private val shouldLoadDistantResources: Boolean,
-    onContactClicked: (contact: Recipient) -> Unit,
-    onDeleteDraftClicked: (message: Message) -> Unit,
-    onDraftClicked: (message: Message) -> Unit,
-    onAttachmentClicked: (attachment: Attachment) -> Unit,
-    onDownloadAllClicked: (message: Message) -> Unit,
-    onReplyClicked: (Message) -> Unit,
-    onMenuClicked: (Message) -> Unit,
-    onAllExpandedMessagesLoaded: () -> Unit,
-    navigateToNewMessageActivity: (Uri) -> Unit,
-    navigateToAttendeeBottomSheet: (List<Attendee>) -> Unit,
-    promptLink: (String, ContextMenuType) -> Unit,
+    private val isForPrinting: Boolean = false,
+    onBodyWebviewFinishedLoading: (() -> Unit)? = null,
+    onContactClicked: ((contact: Recipient) -> Unit)? = null,
+    onDeleteDraftClicked: ((message: Message) -> Unit)? = null,
+    onDraftClicked: ((message: Message) -> Unit)? = null,
+    onAttachmentClicked: ((attachment: Attachment) -> Unit)? = null,
+    onDownloadAllClicked: ((message: Message) -> Unit)? = null,
+    onReplyClicked: ((Message) -> Unit)? = null,
+    onMenuClicked: ((Message) -> Unit)? = null,
+    onAllExpandedMessagesLoaded: (() -> Unit)? = null,
+    navigateToNewMessageActivity: ((Uri) -> Unit)? = null,
+    navigateToAttendeeBottomSheet: ((List<Attendee>) -> Unit)? = null,
+    promptLink: ((String, ContextMenuType) -> Unit)? = null,
```

Comment:
A probably better way than having to carry this around would be to replace all of these callbacks by a single instance of the ThreadAdapterCallbacks object. This list was growing out of hand anyway